### PR TITLE
accessibility assessment of pages using storybook

### PR DESCRIPTION
**Pull request by wenzowski**
This splits the useful work out of #240 so storybook can be used to assess the accessibility of pages according to WCAG 2.0 AA as rendered using mocked data. A static version of the storybook toolchain is published to `/static/storybook/index.html` and should be reviewable in all environments.

### CODE REVIEW

**Comment by wenzowski** [comment]
@dleard the [storyshots enzyme config](https://github.com/bcgov/cas-ciip-portal/blob/02ab1208dfe8241f677ab21814d89c09652a8ef3/app/tests/Storyshots.test.ts) was removed since we've made a team agreement that integration tests should use react-test-renderer to snapshot html output only. If it's needed, or async tests are needed, we can drop it back in.

File: app/.storybook/config.ts [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
-const req = requireContext('../tests', true, /\.stories\.tsx?$/);
+addDecorator(withInfo);
+addDecorator(withCssResources);
+addParameters({
+  cssresources: [{
+    id: 'bootstrap',
+    code: `<link rel="stylesheet" type="text/css" href="/static/bootstrap.min.css"></link>`,
+    picked: true,
+  }]
+})
```
Comment (matthieu-foucault) [inline app/.storybook/config.ts]: This needs some linting. Somehow XO isn't running in the `.storybook` directory

File: app/stories/Pages.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import Admin from 'pages/admin';
+// Import ApplicationReview from 'pages/application-review';
+// import Applications from 'pages/applications';
+import CiipApplicationSWRSImport from 'pages/ciip-application-swrs-import';
+// Import CiipApplication from 'pages/ciip-application';
+// import CompleteSubmit from 'pages/complete-submit';
+// import Index from 'pages/index';
+// import LoginRedirect from 'pages/login-redirect';
+// import OrganisationRequests from 'pages/organisation-requests';
+// import ProductsBenchmarks from 'pages/products-benchmarks';
```
Comment (matthieu-foucault) [inline app/stories/Pages.stories.tsx]: Are this imports here to remind us to test these pages?

File: app/stories/Pages.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      environment={environment}
+      query={query}
+      variables={variables}
+      render={({props, error}) => {
+        if (error) {
+          console.error(error);
+          return 'Error';
+        }
+
+        // Console.log(Component.name, 'props', props);
```
Comment (matthieu-foucault) [inline app/stories/Pages.stories.tsx]:
```suggestion
```

File: app/stories/Pages.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+import {Router} from './mockNextRouter';
+
+const render = Component => {
+  const {query} = Component;
+  const environment = createMockEnvironment();
+  const variables = {...Router.router.query};
+  environment.mock.queueOperationResolver(operation => {
+    const payload = MockPayloadGenerator.generate(operation);
+    // Console.log('mock', operation, payload);
```
Comment (matthieu-foucault) [inline app/stories/Pages.stories.tsx]:
```suggestion
```

File: app/stories/mockNextRouter.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import {UrlObject} from 'url';
+import Router, {Router as RouterType} from 'next/router';
+import {action} from '@storybook/addon-actions';
+
+declare type Url = string | UrlObject;
+
+/* eslint-disable @typescript-eslint/consistent-type-assertions */
```
Comment (matthieu-foucault) [inline app/stories/mockNextRouter.ts]: :scream_cat:

File: app/stories/Pages.stories.tsx (hunk 2 above)
Comment (wenzowski) [reply app/stories/Pages.stories.tsx]: Yes they are (next PR)

File: app/.storybook/config.ts (hunk 1 above)
Comment (wenzowski) [reply app/.storybook/config.ts]: Interesting...I'll look at that

File: app/stories/mockNextRouter.ts (hunk 5 above)
Comment (wenzowski) [reply app/stories/mockNextRouter.ts]: I know, monkeypatching makes typescript sad. I'm not sure how else to mock this though

File: app/.storybook/config.ts (hunk 1 above)
Comment (wenzowski) [reply app/.storybook/config.ts]: Even explicitly calling `xo` with full paths to storybook typescript files, eg. `yarn xo .storybook/*` there are no linter errors. What were you expecting?

File: app/stories/mockNextRouter.ts (hunk 5 above)
Comment (wenzowski) [reply app/stories/mockNextRouter.ts]: omg I'm an idiot. fixed

File: app/.storybook/config.ts (hunk 1 above)
Comment (matthieu-foucault) [reply app/.storybook/config.ts]: Shouldn't have spaces in the curly braces when importing a named import. Missing semicolon in the call to `addParameters`. Extraneous comma in `picked: true,`

**Pull request merged**